src/Events/Editor/Full_Site/Controller.php

```diff
@@ -0,0 +1,407 @@
+<?php
+
+namespace TEC\Events\Editor\Full_Site;
+
+use \Tribe__Events__Main as Events_Main;
+use WP_Block_Template;
+use TEC\Common\Contracts\Provider\Controller as Controller_Contract;
+
+
+/**
+ * Class Controller
+ *
+ * @since   TBD
+ *
+ * @package TEC\Events\Editor\Full_Site
+ */
+class Controller extends Controller_Contract {
+
+	/**
+	 * Register the provider.
+	 *
+	 * @since TBD
+	 */
+	public function do_register(): void {
+		// Register singletons.
+		$this->register_singletons();
+		$this->add_filters();
+		$this->add_actions();
+
+		// Register the Service Provider for Assets.
+		$this->register_assets();
+
+		// Register the service provider itself on the container.
+		$this->container->singleton( static::class, $this );
+	}
+
+	/**
+	 * Unhooks actions and filters.
+	 */
+	public function unregister(): void {
+		$this->remove_actions();
+		$this->remove_filters();
+	}
+
+	/**
+	 * Should only be active if we are in a Site Editor theme.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool Only active during FS theme.
+	 */
+	public function is_active(): bool {
+		return tec_is_full_site_editor();
+	}
+
+	/**
+	 * Registers any requires singletons.
+	 *
+	 * @since 5.14.2
+	 *
+	 */
+	private function register_singletons() {
+		$this->container->singleton( Archive_Events::class, Archive_Events::class, [ 'load' ] );
+		$this->container->singleton( Single_Event::class, Single_Event::class, [ 'load' ] );
+	}
+
+	/**
+	 * Binds and sets up implementations.
+	 *
+	 * @since 5.14.2
+	 */
+	public function register_assets() {
+		$plugin = Events_Main::instance();
+
+		tribe_asset(
+			$plugin,
+			'tec-events-full-site',
+			'app/full-site.js',
```

Comment:
```
			'app/full-site.js',
```
@EugeneKyale I just noticed this. Are we supposed to be using this? Don't see a `full-site.js` anywhere...

Reply:
fyi @bordoni

Reply:
I believe that's a legacy of the level 1 compatibility fixes. It's not currently being used anywhere.